packages/roosterjs-editor-core/lib/editor/Editor.ts

```diff
@@ -359,9 +383,33 @@ export default class Editor {
             content = extractContentEvent.content;
         }
 
+        if (this.core.inDarkMode && !getInDarkMode) {
+            content = this.getLightModeContent(content);
+        }
+
         return content;
     }
 
+    private getLightModeContent(content: string): string {
+        let el = document.createElement('div');
+        el.innerHTML = content;
+        const allChildElements = el.getElementsByTagName('*') as HTMLCollectionOf<HTMLElement>;
+        [].forEach.call(allChildElements, (element: HTMLElement) => {
+            if (element.dataset && (element.dataset.ogsc || element.dataset.ogsb)) {
+                if (element.dataset.ogsc) {
+                    element.style.color = element.dataset.ogsc;
+                }
+
+                if (element.dataset.ogsb) {
+                    element.style.backgroundColor = element.dataset.ogsb;
+                }
+            }
+        });
+        const newContent = el.innerHTML;
+        //document.removeChild(el);

```

Comment:
Should this be removed?

Reply:
1. Consider having a DarkMode plugin, and handle ExtractContentEvent to replace the colors. So that we can add or not add this plugin to support or not support dark mode.
2. Try put all other configurations into this plugin (if we can), so that Editor and its core doesn't need to know too much about dark mode.
3. With this solution, you don't need to add new parameter to getContent()
4. For replacing color, you can leverage HtmlSanitizer from roosterjs-html-sanitizer package and passing in element callback to handle the colors, and the sanitizer code will handle the looping logic for you.